### ISSUE TITLE
Tight list and attributes

### ISSUE THREAD

**Issue by jgm** [jgm/djot#200]
```
{.outer-style}
* a

  {.inner-style}
  * b
* c
```

Ideally the use of the "inner-style" attribute wouldn't force the outer list to be loose.  (If you remove the `{.inner-style}` line, the outer list becomes tight, but the presence of attributes shouldn't affect tight/loose calculations like this.)

**Comment by vassudanagunta** [comment]
My proposal in #201 would solve this conundrum.

**Comment by criloz** [comment]
@jgm I have noticed that when inline attributes are between white space the parser does nothing with them, example https://djot.net/playground/

So a possible solution is to attach the inline attributes within whitespace to their ancestor paragraph, and it will solve this issue too

```djot
* a {.outer-style}
  * b {.inner-style}
* c
```

**Comment by jgm** [comment]
>  to attach the inline attributes within whitespace to their ancestor paragraph

I think this issue is about attaching attributes to lists, not paragraphs within lists.

**Comment by IcedQuinn** [comment]
I ran in to this too. If one hypothetically stores a GTD app's data in a Djot file: you can use attributes to store metadata on tags, but who does the attribute belong to?

The spec suggests block items can have attributes and that list items are defined as block items. Which suggests this:

```
# Project

{review=20260619)
- [ ] Milk
{review=20260619)
- [ ] Eggs
```

But if a list is the result of a series of list items its now ambiguous if the first item's attribute belongs to the entire list or just the first item. And secondly, how would one specify the attribute is for the list item or the whole list in either case?

**Comment by jgm** [comment]
Currently attributes can't go on individual list items.  Perhaps we shouldn't describe individual list items as block elements in the spec, for this reason.  (As you note, there'd be an ambiguity.)

**Comment by guyclaudeburger-collab** [comment]
I suggest needing an indentation putting the the attributes list on the same level as the list item, to differenciate it from the the list's own attributes:
```
{.list-attributes}
     {.list-item-attributes} 
-    list item 1
- list item 2
```
I know we don't like indentation as semantics in general, but sometimes it's the only sound solution.

**Comment by dereuromark** [comment]
@guyclaudeburger-collab For item attr I think the consensus so far was "on the marker itself" - see https://github.com/jgm/djot/issues/250
I dont think this ticket was about that.

### Proposal (based on working solution)
Carve defines looseness by what the item *holds* rather than by whether a blank line appears: an item is loose when it holds a blank-line-separated second **paragraph**. An attached sub-block - block quote, code fence, div, heading, table, or a nested list - leaves the item tight.

So the original input stays tight, with or without the attribute line:

```
{.outer-style}
- a

  {.inner-style}
  - b
- c
```

```html
<ul class="outer-style">
  <li>a
    <ul class="inner-style">
      <li>b</li>
    </ul>
  </li>
  <li>c</li>
</ul>
```

A plain paragraph after the blank is still loose, which is the control:

```
- a

  second paragraph
- c
```

```html
<ul>
  <li><p>a</p>
    <p>second paragraph</p>
  </li>
  <li><p>c</p></li>
</ul>
```

That decoupling is what removes the coupling this issue is about. It was not adopted *for* attributes - it fell out of asking what a blank line inside an item is actually for - but it makes the attribute question independent of tight/loose.

TLDR:

- An attribute block **on its own line** attaches forward to the next block. If that block is a list, the attributes land on the list - including a nested one, which is the case in this issue.
- An attribute block **abutting the marker** (`-{#milk}`, `1.{#step}`) attributes that item.
- A space after the marker means it is not an attribute block at all: `- {.notattr} Milk` renders the braces as text.
